src/main/java/meditracker/command/ViewCommand.java

```diff
@@ -37,9 +58,59 @@ public ViewCommand(String arguments)
      */
     @Override
     public void execute() {
-        String listIndexString = parsedArguments.get(ArgumentName.LIST_INDEX);
-        int listIndex = Integer.parseInt(listIndexString);
+        try {
+            executeFlag();
+            Ui.showSuccessMessage("Medication details has been retrieved");
+
+        } catch (IndexOutOfBoundsException e) {
+            String errorContext = String.format("Invalid medication index specified. %s. " +
+                            "Medicine can not be found", e.getMessage());
+            Ui.showErrorMessage(errorContext);
+
+        } catch (NullPointerException e) {
+            String errorContext = String.format("You have to input a number. %s. " +
+                             "Medicine can not be found", e.getMessage());
+            Ui.showErrorMessage(errorContext);
+
+        } catch (NumberFormatException e) {
+            String errorContext = String.format("Please enter a number. %s. " +
+                            "Medicine can not be found", e.getMessage());
+            Ui.showErrorMessage(errorContext);
+
+        } catch (MedicationNotFoundException e) {
+            String errorContext = String.format("Medicine can not be found. %s. ",
+                    e.getMessage());
+            Ui.showErrorMessage(errorContext);
+
+        }
+    }
+
+    /**
+     * Executes the first flag in the user input.
+     *
+     * @throws MedicationNotFoundException When no medication can be found
+     */
+    private void executeFlag() throws MedicationNotFoundException {
```

Comment:
Might it be better if you check if multiple flags were used and print a warning with `Ui.showWarningMessage` to user that only one of the flag (or the first one rather) will be considered and all remaining will be ignored?